scripts/bundle-image-pickup.sh

```diff
@@ -0,0 +1,187 @@
+
+####################
+## IGNORE VARIABLES
+####################
+IGNORE_API_IMAGE_UPDATE=${IGNORE_API_IMAGE_UPDATE:-"false"}
+IGNORE_COLLECTOR_IMAGE_UPDATE=${IGNORE_COLLECTOR_IMAGE_UPDATE:-"false"}
+IGNORE_INDEXER_IMAGE_UPDATE=${IGNORE_INDEXER_IMAGE_UPDATE:-"false"}
+IGNORE_OPERATOR_IMAGE_UPDATE=${IGNORE_OPERATOR_IMAGE_UPDATE:-"false"}
+IGNORE_POSTGRES_IMAGE_UPDATE=${IGNORE_POSTGRES_IMAGE_UPDATE:-"true"}
+
+####################
+## FUNCTIONS/METHODS
+####################
+
+cleanup () {
+    echo -e "\nRemoving $ORG/$PIPELINE_REPO repository..."
+    rm -rf $PIPELINE_REPO
+}
+
+display_component_images () {
+    echo -e "Component Images"
+    echo -e "==============================================================================" \
+    "\nPOSTGRES:\t\t${POSTGRES_IMAGE:-$DEFAULT_POSTGRES_IMAGE}" \
+    "\nSEARCH_API:\t\t${API_IMAGE:-$DEFAULT_API_IMAGE}" \
+    "\nSEARCH_COLLECTOR:\t${COLLECTER_IMAGE:-$DEFAULT_COLLECTOR_IMAGE}" \
+    "\nSEARCH_INDEXER:\t\t${INDEXER_IMAGE:-$DEFAULT_INDEXER_IMAGE}" \
+    "\nSEARCH_OPERATOR:\t${OPERATOR_IMAGE:-$DEFAULT_OPERATOR_IMAGE}" \
+    "\n==============================================================================\n"
+}
+
+get_default_images_from_csv () {
+  log_color purple "Fetching default component images from ${OPERATOR_CSV_FILEPATH}\n"
+
+  for IMG in $(yq e $OPERATOR_IMAGE_PATH $OPERATOR_CSV_FILEPATH); do
+    if [[ $IMG =~ .*"search-v2-operator".* ]]; then
+      DEFAULT_OPERATOR_IMAGE=$IMG
+    fi
+  done
+
+  for IMG in $(yq e $OPERATOR_ENV_PATH $OPERATOR_CSV_FILEPATH); do
+    if [[ $IMG =~ .*"postgres".* ]]; then
+      DEFAULT_POSTGRES_IMAGE=$IMG
+
+    elif [[ $IMG =~ .*"search-collector".* ]]; then
+      DEFAULT_COLLECTOR_IMAGE=$IMG
+
+    elif [[ $IMG =~ .*"search-indexer".* ]]; then
+      DEFAULT_INDEXER_IMAGE=$IMG
+
+    elif [[ $IMG =~ .*"search-v2-api".* ]]; then
+      DEFAULT_API_IMAGE=$IMG
+    fi
+  done
+
+  display_component_images
+}
+
+update_images_csv () {
+  log_color purple "Preparing to update component: $1 => $2\n"
+
+  # TODO: Replace yq path with $OPERATOR_ENV_PATH. (Note: Adding the env variable seems to cause yq to return no results)
+  if [[ $1 =~ .*"search-indexer".* ]]; then
+    INDEXER_IMAGE=$2
+    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[1].value = "'$INDEXER_IMAGE'"' $OPERATOR_CSV_FILEPATH
+
+  elif [[ $1 =~ .*"search-collector".* ]]; then
+    COLLECTER_IMAGE=$2
+    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[2].value = "'$COLLECTER_IMAGE'"' $OPERATOR_CSV_FILEPATH
+
+  elif [[ $1 =~ .*"search-v2-api".* ]]; then
+    API_IMAGE=$2
+    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[3].value = "'$API_IMAGE'"' $OPERATOR_CSV_FILEPATH
+    
+  elif [[ $1 =~ .*"search-v2-operator".* ]]; then
+    OPERATOR_IMAGE=$2
+    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].image = "'$OPERATOR_IMAGE'"' $OPERATOR_CSV_FILEPATH
+  fi 
+}
```

Comment:
Small suggestion to make the script parameters more clear.
```suggestion
update_images_csv () {
  COMPONENT=$1
  NEW_IMAGE=$2
  log_color purple "Preparing to update component: $COMPONENT => $NEW_IMAGE\n"

  # TODO: Replace yq path with $OPERATOR_ENV_PATH. (Note: Adding the env variable seems to cause yq to return no results)
  if [[ $COMPONENT =~ .*"search-indexer".* ]]; then
    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[1].value = "'$NEW_IMAGE'"' $OPERATOR_CSV_FILEPATH

  elif [[ $COMPONENT =~ .*"search-collector".* ]]; then
    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[2].value = "'$NEW_IMAGE'"' $OPERATOR_CSV_FILEPATH

  elif [[ $COMPONENT =~ .*"search-v2-api".* ]]; then
    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[3].value = "'$NEW_IMAGE'"' $OPERATOR_CSV_FILEPATH
    
  elif [[ $COMPONENT =~ .*"search-v2-operator".* ]]; then
    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].image = "'$NEW_IMAGE'"' $OPERATOR_CSV_FILEPATH
  fi 
}
```

Reply:
Added the suggested changes!